src/lib.rs

```diff
@@ -104,6 +104,21 @@ impl Fernet {
             .duration_since(time::UNIX_EPOCH)
             .unwrap()
             .as_secs();
+        self.encrypt_at_time(data, current_time)
+    }
+
+    /// Encrypts data with the current_time. Returns a value (which is base64-encoded) that can be
+    /// passed to `Fernet::decrypt`.
+    ///
+    /// This function has the capacity to be used incorrectly or insecurely due to
+    /// to the "current_time" parameter. current_time must be the systems time::SystemTime::now()
+    /// with duraction_since(time::UNIX_EPOCH) as seconds.
+    ///
+    /// The motivation for a function like this is for your application to be able to test
+    /// ttl expiry of tokens in your API. This allows you to pass in mock time data to assert
+    /// correct behaviour of your application. Care should be taken to ensure you always pass in
+    /// correct current_time values for deployments.
+    pub fn encrypt_at_time(&self, data: &[u8], current_time: u64) -> String {
```

Comment:
I get the concept behind this, but I do wonder if we're offering folks a foot gun here.

Would it be better to have these as private to the crate, with 

```rust
#[cfg(test)]
{
   pub fn test_encrypt_at_time(&self, data: &[u8], time: u64) -> String {
        self.encrypt_at_time(data, time)
   }
   pub fn test_decrypt_at_time(&self, data: &[u8], time: u64) -> String {
        self.decrypt_at_time(data, time)
   }
}
```
so that the functions are only visible when in a test configuration?

Reply:
Not really, these are needed for downstream testing as well. Using a cfg flag was proposed in the original issue.

Reply:
Yes, I just read https://github.com/mozilla-services/fernet-rs/issues/36#issuecomment-537292631. I'm still a bit concerned because I can see this leading to folks possibly cracking older fernet encryption that should no longer be readable. (Yes, I realize that they could just fork the code if they were intent on doing that, but if that wasn't the intent of the app owner, that path is now a vector of attack.)

Keeping this code limited to explicit uses (e.g. a "test" config or a special build flag) reduces the chance of a future CVE.

Reply:
I wouldn't say cracking, but mis-handling older tokens that *should* have expired. But it's also important for testing that you can handle the timestamps externally to assert confidence you really are doing the right thing when a token expires. I'm currently writing a piece of IDM software, and I really need to be able to show confidence that "yes, when fernet tokens expire I'm handling that properly" and I don't feel like putting in sleep statements to my tests. 

I think the feature flag so far seems like the best option here.